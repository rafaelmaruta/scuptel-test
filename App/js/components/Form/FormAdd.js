import React from 'react';

function FormAdd(props) {
	return <a href="javascript:void(0);" className="form__plans-add" onClick={props.clickAdd}>+</a>
}

export default FormAdd;