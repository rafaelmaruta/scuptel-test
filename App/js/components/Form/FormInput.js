import React from 'react';

const defaultProps = {
	disabled: '',
	onChange: '',
	type    : 'text'
}

function FormInput(props) {
	return (
		<input
			className="form__plans-input"
			disabled={props.disabled}
			min="0"
			onKeyUp={props.onChange}
			onClick={props.onChange}
			placeholder={props.placeholder}
			type={props.type}
			value={props.val}
		/>
	)
}

FormInput.defaultProps = defaultProps;

export default FormInput;