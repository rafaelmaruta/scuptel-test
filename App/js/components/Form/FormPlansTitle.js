import React from 'react';

function FormPlansTitle(props) {
	return <p className="form__plans-title">{props.title}</p>
}

export default FormPlansTitle;