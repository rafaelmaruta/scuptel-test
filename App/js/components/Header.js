import React from 'react';

const defaultProps = {
	title: 'Scup Fale+',
};

function Header(props) {
	return (
		<header className="header">
			<div className="header__center center">
				<a href="#" className="header__logo">{props.title}</a>
			</div>
		</header>
	);
}

Header.defaultProps = defaultProps;

export default Header;