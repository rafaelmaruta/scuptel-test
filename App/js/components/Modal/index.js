import React, { PropTypes } from 'react';
import ModalBg from './ModalBg';
import ModalClose from './ModalClose';
import ModalRow from './ModalRow';

const propTypes = {
	details    : PropTypes.array.isRequired,
	modalState : PropTypes.string.isRequired,
	toggleModal: PropTypes.func.isRequired
};

const defaultProps = {
	details    : [],
	modalState : 'hide',
	toggleModal: function() {}
};

function Modal(props) {
	return (
		<section className={"modal " + props.modalState}>
			<ModalBg clickBg={props.toggleModal} />
			<div className="modal__popup">
				<ModalClose clickClose={props.toggleModal} />
				<table className="modal__popup-table">
					<tbody>
						{props.details.map((data, index) => (
							<ModalRow
								city={data.city}
								ddd={data.ddd}
								fillInput={props.fillInput}
								key={index}
							/>
						))}
					</tbody>
				</table>
			</div>
		</section>
	)
}

Modal.propTypes = propTypes;
Modal.defaultProps = defaultProps;

export default Modal;