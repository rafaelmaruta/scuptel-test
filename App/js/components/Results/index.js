import React, { PropTypes } from 'react';
import ResultsTitle from './ResultsTitle';
import ResultsValue from './ResultsValue';

const propTypes = {
	plans: PropTypes.array.isRequired
};

const defaultProps = {
	plans: []
};

function Results(props) {
	return (
		<section className="results">
			<div className="results__center">
				{props.plans.map((data, index) => (
					<div className="results__group" key={index}>
						<ResultsTitle title={data.plan} />
						<ResultsValue
							price={props.price}
							time={props.time}
							plan={data.time}
						/>
					</div>
				))}
				<div className="results__group results__group--last">
					<ResultsTitle title="Normal" />
					<ResultsValue
						price={props.price}
						time={props.time}
					/>
				</div>
			</div>
		</section>
	)
}

Results.propTypes    = propTypes;
Results.defaultProps = defaultProps;

export default Results;