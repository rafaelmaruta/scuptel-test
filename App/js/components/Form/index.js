import React from 'react';
import FormTitle from './FormTitle';
import FormPlansTitle from './FormPlansTitle';
import FormInput from './FormInput';
import FormAdd from './FormAdd';

function Form(props) {
	return (
		<section className="form">
			<div className="form__center center">
				<FormTitle />
				<form className="form__plans">
					<div className="form__plans-group">
						<FormPlansTitle title="Origem" />
						<FormInput
							disabled="disabled"
							placeholder="DDD"
							val={props.origin}
						/>
						<FormAdd clickAdd={() => props.toggleModal('show', 'origin')} />
					</div>
					<div className="form__plans-group">
						<FormPlansTitle title="Destino" />
						<FormInput
							disabled="disabled"
							placeholder="DDD"
							val={props.destiny}
						/>
						<FormAdd clickAdd={() => props.toggleModal('show', 'destiny')} />
					</div>
					<div className="form__plans-group form__plans-group--last">
						<FormPlansTitle title="Tempo (min)" />
						<FormInput
							placeholder="MM"
							type="number"
							onChange={props.setTime}
						/>
					</div>
				</form>
			</div>
		</section>
	)
}

export default Form;