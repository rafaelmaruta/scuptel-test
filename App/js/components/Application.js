import jQuery from 'jquery';
import React from 'react';
import Header from './Header';
import Form from './Form';
import Results from './Results';
import Modal from './Modal';
import RequestsAsync from './Utils/RequestsAsync';

// Main stateful component
class Application extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			pricing   : this.props.pricing,
			details   : this.props.details,
			plans     : this.props.plans,
			origin    : '',
			destiny   : '',
			modalMode : '',
			modalState: 'hide',
			price     : undefined,
			testing   : this.props.testing,
			time      : 0
		}
		this.fillData    = this.fillData.bind(this);
		this.fillInput   = this.fillInput.bind(this);
		this.setPrice    = this.setPrice.bind(this);
		this.setTime     = this.setTime.bind(this);
		this.toggleModal = this.toggleModal.bind(this);
	}
	componentDidMount() {
		if (this.props.testing !== 'true') {
			RequestsAsync(this.fillData);
		} else {
			this.fillData({
				pricing: this.props.pricing,
				details: this.props.details,
				plans  : this.props.plans
			});
		};
	}
	fillData(results) {
		this.setState({
			pricing: results.pricing[0],
			details: results.details[0],
			plans  : results.plans[0]
		})
	}
	fillInput(ddd) {
		if (this.state.modalMode === 'origin') {
			this.setState({origin: ddd}, () => this.setPrice());
		} else if (this.state.modalMode === 'destiny') {
			this.setState({destiny: ddd}, () => this.setPrice());
		};
		this.toggleModal('hide');
	}
	setPrice() {
		if (this.state.pricing.data) {
			var obj = this.state.pricing.data.filter(price => {
				return price.origin === this.state.origin && price.destiny === this.state.destiny
			});

			if (obj.length) {
				this.setState({price: obj[0].price});
			} else {
				this.setState({price: undefined});
			};
		};
	}
	setTime(event) {
		this.setState({time: event.target.value}, () => this.setPrice());
	}
	toggleModal(state, mode) {
		mode = mode || '';
		this.setState({
			modalState: state,
			modalMode : mode
		});
	}
	render() {
		var plans  = this.state.plans !== undefined ? this.state.plans : {data: []},
			details = this.state.details !== undefined ? this.state.details : {data: []};

		return (
			<div>
				<Header title="Scup Fale+" />
				<Form
					destiny={this.state.destiny}
					origin={this.state.origin}
					setTime={this.setTime}
					toggleModal={this.toggleModal}
				/>
				<Results
					plans={plans.data}
					price={this.state.price}
					time={this.state.time}
				/>
				<Modal
					details={details.data}
					fillInput={this.fillInput}
					modalState={this.state.modalState}
					toggleModal={this.toggleModal}
				/>
			</div>
		)
	}
}

export default Application;