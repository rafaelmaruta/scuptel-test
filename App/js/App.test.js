import React from 'react';
import ReactTestUtils from 'react-addons-test-utils';
//import renderer from 'react-test-renderer';
import jsdom from 'jsdom';
import Application from './components/Application';
import Header from './components/Header';
import Form from './components/Form';
import Results from './components/Results';
import Modal from './components/Modal';
import FormatPrice from './components/Utils/Helper';


const pricing = {
    total: 6,
    data: [
        {
            origin: "011",
            destiny: "016",
            price: "1.90"
        },
        {
            origin: "016",
            destiny: "011",
            price: "2.90"
        },
        {
            origin: "011",
            destiny: "017",
            price: "1.70"
        },
        {
            origin: "017",
            destiny: "011",
            price: "2.70"
        },
        {
            origin: "011",
            destiny: "018",
            price: "0.90"
        },
        {
            origin: "018",
            destiny: "011",
            price: "1.90"
        }
    ]
};

const plans = {
    total: 3,
    data: [
        {
            plan: "Test 30",
            time: "30"
        },
        {
            plan: "Test 60",
            time: "60"
        },
        {
            plan: "Test 120",
            time: "120"
        }
    ]
};

const details = {
    total: 4,
    data: [
        {
			ddd : "011",
			city: "Test1"
        },
        {
			ddd : "016",
			city: "Test2"
        },
        {
			ddd : "017",
			city: "Test3"
        },
        {
			ddd : "018",
			city: "Test4"
        }
    ]
};

describe('Methods', () => {

	it('format price works', () => {
		var price = FormatPrice.getSplit(1.2345),
			priceSplit = {
				int  : '1',
				float: '23'
			};

		expect(price).toEqual(priceSplit);
	});
});

describe('Event', () => {

	beforeEach(function() {
	    this.renderer = ReactTestUtils.createRenderer();
	    this.renderer.render(
	    	<Application
				testing="true"
				pricing={pricing}
				details={details}
				plans={plans}
			/>
		);

	    this.TestUtils = ReactTestUtils;
	});

	it("is a list element", function() {
	    let renderedRoot = this.renderer.getRenderOutput();
	    console.log(renderedRoot.props.children[1].refs.clickAdd);

	    //expect(renderedRoot.type).toEqual("li");
	});


});

describe('Component', () => {

	const application = renderer.create(
		<Application
			testing="true"
			pricing={pricing}
			details={details}
			plans={plans}
		/>
	);

	it('application rendering works', () => {
		console.log('application', application.props.children);
		let tree = application.toJSON();
		expect(tree).toMatchSnapshot();
	});

	const header = renderer.create(
		<Header title="Teste Header" />
	);

	it('header rendering works', () => {

		let tree = header.toJSON();
		expect(tree).toMatchSnapshot();
	});

	const form = renderer.create(
		<Form
			destiny="011"
			origin="011"
			setTime="10"
		/>
	);

	it('form rendering works', () => {

		let tree = form.toJSON();
		expect(tree).toMatchSnapshot();
	});

	const results = renderer.create(
		<Results
			plans={plans.data}
			price="2.7"
			time="61"
		/>
	);
	
	it('results rendering works', () => {

		let tree = results.toJSON();
		expect(tree).toMatchSnapshot();
	});

	const modal = renderer.create(
		<Modal
			details={details.data}
			modalState="show"
		/>
	);
	
	it('modal rendering works', () => {
		let tree = modal.toJSON();
		expect(tree).toMatchSnapshot();
	});

});

//testes:

// Carregar JSONs
// Clique dos botões +
// Clique na hora de selecionar o plano
// Clique na hora de fechar o modal
// Plano existente está preenchendo resultado
// Plano inexistente não está preenchendo nada
// Verificar se os input text estão preenchidos