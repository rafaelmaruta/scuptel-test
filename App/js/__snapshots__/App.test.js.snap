exports[`Component form rendering works 1`] = `
<section
  className="form">
  <div
    className="form__center center">
    <p
      className="form__title">
      Comparar Planos
    </p>
    <form
      className="form__plans">
      <div
        className="form__plans-group">
        <p
          className="form__plans-title">
          Origem
        </p>
        <input
          className="form__plans-input"
          disabled="disabled"
          min="0"
          onClick=""
          onKeyUp=""
          placeholder="DDD"
          type="text"
          value="011" />
        <a
          className="form__plans-add"
          href="javascript:void(0);"
          onClick={[Function]}
          refs="clickAdd">
          +
        </a>
      </div>
      <div
        className="form__plans-group">
        <p
          className="form__plans-title">
          Destino
        </p>
        <input
          className="form__plans-input"
          disabled="disabled"
          min="0"
          onClick=""
          onKeyUp=""
          placeholder="DDD"
          type="text"
          value="011" />
        <a
          className="form__plans-add"
          href="javascript:void(0);"
          onClick={[Function]}
          refs="clickAdd">
          +
        </a>
      </div>
      <div
        className="form__plans-group form__plans-group--last">
        <p
          className="form__plans-title">
          Tempo (min)
        </p>
        <input
          className="form__plans-input"
          disabled=""
          min="0"
          onClick="10"
          onKeyUp="10"
          placeholder="MM"
          type="number"
          value={undefined} />
      </div>
    </form>
  </div>
</section>
`;

exports[`Component header rendering works 1`] = `
<header
  className="header">
  <div
    className="header__center center">
    <a
      className="header__logo"
      href="#">
      Teste Header
    </a>
  </div>
</header>
`;

exports[`Component modal rendering works 1`] = `
<section
  className="modal show">
  <div
    className="modal__bg modal__bg--dark"
    onClick={[Function]} />
  <div
    className="modal__popup">
    <a
      className="modal__popup-close"
      href="javascript:void(0);"
      onClick={[Function]}>
      x
    </a>
    <table
      className="modal__popup-table">
      <tbody>
        <tr
          onClick={[Function]}>
          <td>
            011
             - 
            Test1
          </td>
        </tr>
        <tr
          onClick={[Function]}>
          <td>
            016
             - 
            Test2
          </td>
        </tr>
        <tr
          onClick={[Function]}>
          <td>
            017
             - 
            Test3
          </td>
        </tr>
        <tr
          onClick={[Function]}>
          <td>
            018
             - 
            Test4
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</section>
`;

exports[`Component results rendering works 1`] = `
<section
  className="results">
  <div
    className="results__center">
    <div
      className="results__group">
      <p
        className="results__group-title">
        Test 30
      </p>
      <p
        className="results__group-value">
        R$
        <span>
          92
        </span>
        ,
        07
      </p>
    </div>
    <div
      className="results__group">
      <p
        className="results__group-title">
        Test 60
      </p>
      <p
        className="results__group-value">
        R$
        <span>
          2
        </span>
        ,
        97
      </p>
    </div>
    <div
      className="results__group">
      <p
        className="results__group-title">
        Test 120
      </p>
      <p
        className="results__group-value">
        R$
        <span>
          0
        </span>
        ,
        00
      </p>
    </div>
    <div
      className="results__group results__group--last">
      <p
        className="results__group-title">
        Normal
      </p>
      <p
        className="results__group-value">
        R$
        <span>
          164
        </span>
        ,
        70
      </p>
    </div>
  </div>
</section>
`;
